base/compiler/abstractinterpretation.jl

```diff
@@ -2544,6 +2544,7 @@ end
 function abstract_eval_statement(interp::AbstractInterpreter, @nospecialize(e), vtypes::VarTable, sv::InferenceState)
     if !isa(e, Expr)
         if isa(e, PhiNode)
+            add_curr_ssaflag!(sv, IR_FLAG_EFFECT_FREE)
```

Comment:
Why is this needed? This function works on `InferenceState` and thus irinterp will never use it. The ordinary inference may see PhiNode somehow?

Reply:
We made it legal in untyped code, but it is true that we may need to add the flag in ssa conversion also.

Reply:
I wonder if there is any actual users of this `PhiNode` handling within untyped code inference. And it sounds a good idea to add the flag during SSA conversion.

Reply:
> I wonder if there is any actual users of this PhiNode handling within untyped code inference.
Diffractor uses it.

> And it sounds a good idea to add the flag during SSA conversion.
I think we already set this since `stmt_effect_flags` knows about it, but we've of course been moving to not calling that function quite as often.